app/scripts/views/settings/communication_preferences.js

```diff
@@ -89,39 +95,43 @@ define(function (require, exports, module) {
     },
 
     submit () {
-      var emailPrefs = this.getMarketingEmailPrefs();
-      return this.setOptInStatus(NEWSLETTER_ID, ! emailPrefs.isOptedIn(NEWSLETTER_ID));
+      return this._optIn();
     },
 
-    setOptInStatus (newsletterId, isOptedIn) {
-      let eventPrefix;
-      let method;
-
-      if (isOptedIn) {
-        eventPrefix = '';
-        method = 'optIn';
-      } else {
-        eventPrefix = 'un';
-        method = 'optOut';
-      }
-
-      this.logViewEvent(method);
-
-      return this.getMarketingEmailPrefs()[method](newsletterId)
+    /**
+     * Opt in to the marketing email
+     *
+     * @returns {Promise}
+     * @private
+     */
+    _optIn () {
```

Comment:
I tried to simplify this logic a bit by using two different functions, one for opting in the user, the other to allow the user to manage their account.